File: 27_remove_element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        left = 0
        right = len(nums) - 1
        counter = 0
        while left <= right:
            if nums[left] == val:
                swapped = nums[right]
                nums[right] = nums[left]
                nums[left] = swapped
                right -= 1
                counter += 1
            else:
                left += 1
        return len(nums[:len(nums) - counter])

File: 27_remove_element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int left = 0;
        int right = nums.size() - 1;
        int counter = 0;
        int swapped = 0;
        while (left <= right) {
            if (nums[left] == val) {
                swapped = nums[right];
                nums[right] = nums[left];
                nums[left] = swapped;
                right -= 1;
                counter += 1;
            } else {
                left += 1;
            }
        }
        std::vector<int> sliced(nums.begin(), nums.begin() + (nums.size() - counter));
        return sliced.size();
    }
};